### PR TITLE
Document rule board

### PR DESCRIPTION
**Pull request by underyx**
Fixes PD-271

### Security

- [ ] Change has security implications (if so, ping the security team)

### CODE REVIEW

File: docs/semgrep-app/managing-policy.md [hunk 1]
```diff
@@ -7,6 +7,21 @@ import MoreHelp from "/src/components/MoreHelp"
...
+  Policy management is an advanced feature
+  for orchestrating complex
+  rule and notification channel assignments.
+  
+  | **Semgrep App Tier** | **Signup date** | **Availability** |
+  |----------|-----------------|------------------|
+  | Enterprise     | Any             | available        |
+  | Team     | Any             | available        |
+  | Community     | before September 15, 2021             | available        |
+  | Community     | after September 15, 2021             | unavailable, use the [Rule board](rule-board.md) instead.|
```
Comment (pabloest) [inline docs/semgrep-app/managing-policy.md:21]: Could the table be replaced by one or two sentences?

For example:

This feature is available to all users on the Enterprise and Team tiers, and Community tier users who signed up before September 15, 2021. Community tier users who signed up after September 15, 2021 should use the Rule board instead.

I _think_ that’s a bit easier to parse.

File: docs/semgrep-app/rule-board.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+---
+slug: rule-board
+append_help_link: true
+---
+
+# Rule board
+
+The rule board is a visual representation of the rules
+that Semgrep uses to identify potential security issues.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:9]:
```suggestion
The rule board is a visual representation of the rules
that Semgrep App uses to scan code in CI.
```
^^ suggesting the above to not limit the use case to security issues

File: docs/semgrep-app/rule-board.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+Rules and rulesets are displayed as cards,
+which you can drag and drop between columns.
+When a Semgrep scan runs in CI,
+the patterns it scans for
+and the actions it takes in response to findings
+are determined by where cards are placed on the rule board.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:16]:
```suggestion
The placement of cards in columns determines the Semgrep rules that run in CI and the actions taken in response to findings.
```

File: docs/semgrep-app/rule-board.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+:::tip
+
+  To show results to developers in a non-blocking manner,
+  you can set this column to notify via PR comments.
+
+:::
+
+### Block column
+
+Findings from cards of this column are shown to developers as blocking in CI.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:37]:
```suggestion
Findings from cards placed in this column are shown to developers as blocking in CI.
```

File: docs/semgrep-app/rule-board.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+This means these findings need to be addressed before merging the PR.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:38]:
```suggestion
These findings need to be addressed before merging the PR.
```

File: docs/semgrep-app/rule-board.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+#### Registry search
+
+Search the Semgrep Registry for rulesets via the search box at the top of the gutter.
+Search results will show up in this column.
+
+#### Recommended cards
+
+Semgrep's rule recommendation engine will show ruleset cards in the gutter
+when it finds technologies in your repositories with relevant rulesets available.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:52]:
```suggestion
Semgrep will recommend relevant rulesets based on technologies it finds in your repositories and show them here.
```

File: docs/semgrep-app/rule-board.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+#### Disabled rules
+
+If you removed a rule from an active ruleset,
+you can find the rule card and restore it from here.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:57]:
```suggestion
Find and restore rules previously removed from active rulesets here.
```

File: docs/semgrep-app/rule-board.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+### Rulesets by name
+
+Use the search box in the gutter column to find specific rulesets.
+Rulesets are usually named after technologies and vulnerability classes.
+
+:::tip
+
+  If you're unsure about a ruleset,
+  you can always expand it and drag just some of its rules onto the board.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:76]:
```suggestion
  If you're unsure about a ruleset,
  try expanding it and dragging just some of its rules onto the board.
```

File: docs/semgrep-app/rule-board.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+There are various ways to add cards to your rule board.
+
+### Recommended rulesets
+
+The gutter displays recommended rulesets for technologies in your repositories.
+We suggest you add all these cards to the Audit column as they show up.
+
+### Rulesets by name
+
+Use the search box in the gutter column to find specific rulesets.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:70]:
```suggestion
Use the search in the gutter column to find specific rulesets.
```

File: docs/semgrep-app/rule-board.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+This means these findings need to be addressed before merging the PR.
+
+### Gutter
+
+The gutter on the right side of the board displays various kinds of cards that are inactive.
+
+#### Registry search
+
+Search the Semgrep Registry for rulesets via the search box at the top of the gutter.
+Search results will show up in this column.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:47]:
```suggestion
Search results will appear in this column.
```

File: docs/semgrep-app/rule-board.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+  you can always expand it and drag just some of its rules onto the board.
+
+:::
+
+### Rules from the registry
+
+As you explore the [Semgrep Registry](https://semgrep.dev/r),
+you will see buttons named "Add these to Policy".
+Click this button and select the "Rule board" option.
+The new cards will show up on your board when you next open the page.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:85]:
```suggestion
The new cards will appear on your board when you next open the page.
```

File: docs/semgrep-app/rule-board.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+To remove a ruleset, just drag it to the gutter on the right.
+
+### Rules
+
+To disable just one rule from a ruleset,
+expand the ruleset with the "rules" button.
+You can then drag specific rules to the gutter on the right.
+This will disable those rules but keep the rest of the ruleset enabled.
+
+## Setting up notifications
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:107]:
```suggestion
## Configuring notifications
```

File: docs/semgrep-app/rule-board.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+
+To configure notifications,
+click the bell button in the top right corner of a column.
```
Comment (pabloest) [inline docs/semgrep-app/rule-board.md:110]: What can users do here? I wasn’t able to try this in our rule board (perhaps I don’t have the right permissions).

File: docs/semgrep-app/managing-policy.md (hunk 1 above)
Comment (chmccreery) [reply docs/semgrep-app/managing-policy.md:21]: I agree 😄

Comment (underyx) [reply docs/semgrep-app/managing-policy.md:21]: heey this is what I was asking you about last Friday :D

Comment (pabloest) [reply docs/semgrep-app/managing-policy.md:21]: 👀

File: docs/semgrep-app/rule-board.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+are determined by where cards are placed on the rule board.
+
+![Screenshot of the default state of the rule board](../img/rule-board.png)
+
+## Columns
+
+### Audit column
+
+Findings from cards of this column are hidden from developers by default.
+They will not block merging pull requests or be reported in the code review process.
```
[the remark this answers is not in the document]
Comment (underyx) [reply docs/semgrep-app/rule-board.md:25]: Either PR comments, or CI job logs, or just the red :x: on a PR.

File: docs/semgrep-app/rule-board.md [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+  you can set this column to notify via PR comments.
+
+:::
+
+### Block column
+
+Findings from cards of this column are shown to developers as blocking in CI.
+This means these findings need to be addressed before merging the PR.
+
+### Gutter
```
[the remark this answers is not in the document]
Comment (underyx) [reply docs/semgrep-app/rule-board.md:40]: I think sidebar is taken already by the navigation sidebar on the left :/

[the remark this answers is not in the document]
Comment (underyx) [reply docs/semgrep-app/rule-board.md:40]: What do you call notes that are handwritten on the margin of a physical book? 🤔

[the remark this answers is not in the document]
Comment (underyx) [reply docs/semgrep-app/rule-board.md:37]: BTW the concept I was thinking of was the gutter on the a bowling lane's side. Or on the side of a roof.

File: docs/semgrep-app/rule-board.md (hunk 13 above)
Comment (underyx) [reply docs/semgrep-app/rule-board.md:105]: Added 

```
You can enable [third-party notifications](../notifications/)
and [inline pull request comments](../notifications/#pull-request-comments)
on a per-column basis in the resulting panel.
```

**Review by pabloest — APPROVED** [review]

**Pull request merged**